### PR TITLE
[IMP] website_slides: archive slide section from frontend

### PR DESCRIPTION
**Pull request by sga-odoo**
**PURPOSE:**

- Let the user delete a section in frontend

**SPECIFICATION:**

- Add the trash icon at the same level of the section. 
- SHOW on hover:  https://tinyurl.com/yy5u9m99
- If section not empty, move the content under un-categorised.

**LINKS:**
TaskID: 2093671
Closes: https://github.com/odoo/odoo/pull/49646

### CODE REVIEW

File: addons/website_slides/controllers/main.py [hunk 1]
```diff
@@ -794,6 +794,22 @@ def slide_archive(self, slide_id):
 
         return False
 
+    @http.route('/slides/category/archive', type='json', auth='user', website=True)
+    def slide_category_archive(self, category_id):
+        """ This route allows channel publishers to archive categories.
+        It has to be done in sudo mode since only website_publishers can write on slides in ACLs """
+        category = request.env['slide.slide'].browse(category_id)
```
Comment (tde-banana-odoo) [inline addons/website_slides/controllers/main.py:801]: Just to be sure, make an exists() and return a right error if not found.

File: addons/website_slides/controllers/main.py [hunk 2]
```diff
@@ -794,6 +794,22 @@ def slide_archive(self, slide_id):
...
+        slide_ids = category.slide_ids
```
Comment (tde-banana-odoo) [inline addons/website_slides/controllers/main.py:802]: Linting: that variable is not really necessary, is it ?

File: addons/website_slides/controllers/main.py [hunk 3]
```diff
@@ -794,6 +794,22 @@ def slide_archive(self, slide_id):
...
+        if category.channel_id.can_publish:
+            for slide in slide_ids.sudo():
+                slide.category_id = False
+                category.channel_id.sudo()._resequence_slides(slide)
```
Comment (tde-banana-odoo) [inline addons/website_slides/controllers/main.py:806]: What happens if you don't resequence for each slide but once after removing category_id of all slides ? Is it crashing ?

File: addons/website_slides/models/slide_slide.py [hunk 4]
```diff
@@ -219,7 +219,7 @@ def _compute_category_id(self):
         self.category_id = False  # initialize whatever the state
 
         channel_slides = {}
-        for slide in self:
+        for slide in self.with_context(active_test=True):
```
Comment (tde-banana-odoo) [inline addons/website_slides/models/slide_slide.py:222]: Mmh ... why this fix ? As _compute_category_id is called by the ORM on a record set, it is impacted by this flag ?

File: addons/website_slides/models/slide_slide.py [hunk 5]
```diff
@@ -279,7 +279,7 @@ def _compute_slide_views(self):
         for slide in self:
             slide.slide_views = mapped_data.get(slide.id, 0)
 
-    @api.depends('slide_ids.sequence', 'slide_ids.slide_type', 'slide_ids.is_published', 'slide_ids.is_category')
+    @api.depends('channel_id.slide_ids.sequence', 'channel_id.slide_ids.slide_type', 'channel_id.slide_ids.is_published', 'channel_id.slide_ids.is_category')
```
Comment (tde-banana-odoo) [inline addons/website_slides/models/slide_slide.py:282]: Why this change ?

File: addons/website_slides/controllers/main.py (hunk 3 above)
Comment (sga-odoo) [reply addons/website_slides/controllers/main.py:806]: If we just apply category_id = False it doesn't effect sequence of slides. Because category_id recompute based on sequence number. So at some point it will work on initial phase but as soon as we add new un-categorised slide it will spoil the sequence of slides and moved to another category which is not relevant to it. (ref: https://github.com/odoo/odoo/blob/master/addons/website_slides/models/slide_slide.py#L219)

So it requires to update sequence number just like when we are adding new un-categorised slide.

File: addons/website_slides/models/slide_slide.py (hunk 4 above)
Comment (sga-odoo) [reply addons/website_slides/models/slide_slide.py:222]: Yes _compute_category_id is called by the ORM but seems it's fetching inactive slides/category too. Which is not expected behaviour.

File: addons/website_slides/models/slide_slide.py (hunk 5 above)
Comment (sga-odoo) [reply addons/website_slides/models/slide_slide.py:282]: While deleting category/section from backend It throws foreign key constraint. So according to product owner we need default behaviour there. But currently to get slides statistics data relationship is establish using 'slide_ids' field which is related to category_id. So due to wrongly/intended behaviour mapping, it throws constraint error. Rather then fetching from new relationship I have used existing one which is already created using channel_id.

**Comment by tde-banana-odoo** [comment]
@robodoo r+
